pkg/core/secrets/manager/global_manager.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
+	secret_cipher "github.com/kumahq/kuma/pkg/core/secrets/cipher"
+	secret_store "github.com/kumahq/kuma/pkg/core/secrets/store"
+)
+
+func NewGlobalSecretManager(secretStore secret_store.SecretStore, cipher secret_cipher.Cipher) manager.ResourceManager {
+	return &globalSecretManager{
+		secretStore: secretStore,
+		cipher:      cipher,
+	}
+}
+
+var _ manager.ResourceManager = &globalSecretManager{}
+
+type globalSecretManager struct {
+	secretStore secret_store.SecretStore
+	cipher      secret_cipher.Cipher
+}
+
+func (s *globalSecretManager) Get(ctx context.Context, resource model.Resource, fs ...core_store.GetOptionsFunc) error {
+	secret, ok := resource.(*secret_model.GlobalSecretResource)
+	if !ok {
+		return newInvalidTypeError()
+	}
+	if err := s.secretStore.Get(ctx, secret, fs...); err != nil {
+		return err
+	}
+	return s.decrypt(secret)
+}
+
+func (s *globalSecretManager) List(ctx context.Context, resources model.ResourceList, fs ...core_store.ListOptionsFunc) error {
+	secrets, ok := resources.(*secret_model.GlobalSecretResourceList)
+	if !ok {
+		return newInvalidTypeError()
+	}
+	if err := s.secretStore.List(ctx, secrets, fs...); err != nil {
+		return err
+	}
+	for _, secret := range secrets.Items {
+		if err := s.decrypt(secret); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *globalSecretManager) Create(ctx context.Context, resource model.Resource, fs ...core_store.CreateOptionsFunc) error {
+	secret, ok := resource.(*secret_model.GlobalSecretResource)
+	if !ok {
+		return newInvalidTypeError()
+	}
+	if err := s.encrypt(secret); err != nil {
+		return err
+	}
+	if err := s.secretStore.Create(ctx, secret, append(fs, core_store.CreatedAt(time.Now()))...); err != nil {
+		return err
+	}
+	return s.decrypt(secret)
+}
+
+func (s *globalSecretManager) Update(ctx context.Context, resource model.Resource, fs ...core_store.UpdateOptionsFunc) error {
+	secret, ok := resource.(*secret_model.GlobalSecretResource)
+	if !ok {
+		return newInvalidTypeError()
+	}
+	if err := s.encrypt(secret); err != nil {
+		return err
+	}
+	if err := s.secretStore.Update(ctx, secret, append(fs, core_store.ModifiedAt(time.Now()))...); err != nil {
+		return err
+	}
+	return s.decrypt(secret)
+}
+
+func (s *globalSecretManager) Delete(ctx context.Context, resource model.Resource, fs ...core_store.DeleteOptionsFunc) error {
+	secret, ok := resource.(*secret_model.GlobalSecretResource)
+	if !ok {
+		return newInvalidTypeError()
+	}
+	return s.secretStore.Delete(ctx, secret, fs...)
+}
+
+func (s *globalSecretManager) DeleteAll(ctx context.Context, secrets model.ResourceList, fs ...core_store.DeleteAllOptionsFunc) error {
+	list := &secret_model.SecretResourceList{}
+	opts := core_store.NewDeleteAllOptions(fs...)
+	if err := s.secretStore.List(context.Background(), list, core_store.ListByMesh(opts.Mesh)); err != nil {
```

Comment:
I'm pretty sure it will work, but I'd better get rid of any mentioning of `opts.Mesh` in the global secret manager. Maybe we can replace `opts.Mesh` with `core_model.NoMesh` here?

Reply:
good point. Removed `ListByMesh` althogether (which is equivalent of `ListByMesh(NoMesh)`)